py-polars/polars/internals/dataframe/frame.py

```diff
@@ -274,20 +274,20 @@ def __init__(
         elif isinstance(data, dict):
             self._df = dict_to_pydf(data, columns=columns)
 
-        elif _NUMPY_TYPE(data) and isinstance(data, np.ndarray):
-            self._df = numpy_to_pydf(data, columns=columns, orient=orient)
-
-        elif _PYARROW_TYPE(data) and isinstance(data, pa.Table):
-            self._df = arrow_to_pydf(data, columns=columns)
-
         elif isinstance(data, Sequence) and not isinstance(data, str):
             self._df = sequence_to_pydf(
                 data, columns=columns, orient=orient, infer_schema_length=50
             )
         elif isinstance(data, pli.Series):
             self._df = series_to_pydf(data, columns=columns)
 
-        elif _PANDAS_TYPE(data) and isinstance(data, pd.DataFrame):
+        elif _check_for_numpy(data) and isinstance(data, np.ndarray):
+            self._df = numpy_to_pydf(data, columns=columns, orient=orient)
+
+        elif _check_for_pyarrow(data) and isinstance(data, pa.Table):
+            self._df = arrow_to_pydf(data, columns=columns)
+
+        elif _check_for_pandas(data) and isinstance(data, pd.DataFrame):
             self._df = pandas_to_pydf(data, columns=columns)
 
         elif isinstance(data, (Generator, Iterable)) and not isinstance(data, Sized):
```

Comment:
Switching the condition here might also be faster (not sure how common it is to have Sixed):
```elif not isinstance(data, Sized) and isinstance(data, (Generator, Iterable))```

Reply:
Makes sense to me; done.